source/extensions/stat_sinks/metrics_service/grpc_metrics_service_impl.cc

```diff
@@ -60,21 +60,39 @@ void MetricsServiceSink::flushGauge(const Stats::Gauge& gauge) {
   gauage_metric->set_value(gauge.value());
 }
 
-void MetricsServiceSink::flushHistogram(const Stats::ParentHistogram& histogram) {
-  io::prometheus::client::MetricFamily* metrics_family = message_.add_envoy_metrics();
-  metrics_family->set_type(io::prometheus::client::MetricType::SUMMARY);
-  metrics_family->set_name(histogram.name());
-  auto* metric = metrics_family->add_metric();
-  metric->set_timestamp_ms(std::chrono::duration_cast<std::chrono::milliseconds>(
-                               time_source_.systemTime().time_since_epoch())
-                               .count());
-  auto* summary_metric = metric->mutable_summary();
-  const Stats::HistogramStatistics& hist_stats = histogram.intervalStatistics();
+void MetricsServiceSink::flushHistogram(const Stats::ParentHistogram& envoy_histogram) {
+  // Add summary information for histograms.
+  io::prometheus::client::MetricFamily* summary_metrics_family = message_.add_envoy_metrics();
+  summary_metrics_family->set_type(io::prometheus::client::MetricType::SUMMARY);
+  summary_metrics_family->set_name(envoy_histogram.name());
+  auto* summary_metric = summary_metrics_family->add_metric();
+  summary_metric->set_timestamp_ms(std::chrono::duration_cast<std::chrono::milliseconds>(
+                                       time_source_.systemTime().time_since_epoch())
+                                       .count());
+  auto* summary = summary_metric->mutable_summary();
+  const Stats::HistogramStatistics& hist_stats = envoy_histogram.intervalStatistics();
   for (size_t i = 0; i < hist_stats.supportedQuantiles().size(); i++) {
-    auto* quantile = summary_metric->add_quantile();
+    auto* quantile = summary->add_quantile();
     quantile->set_quantile(hist_stats.supportedQuantiles()[i]);
     quantile->set_value(hist_stats.computedQuantiles()[i]);
   }
+
+  // Add bucket information for histograms.
```

Comment:
I don't have a super strong opinion as I don't know who is using this, but it does seem like this could increase cost a bunch for folks not using this data? If we don't config guard it, can we at least add a TODO to make sending both summary and bucket information configurable? (Since with bucket information, the summary information can be easily computed)

Reply:
Sure added a TODO. The main reason I was not doing config change is to avoid it if it is not needed and if some one finds this to be a huge perf issue, we can add it later.